lib/saluki-components/src/sources/otlp/mod.rs

```diff
@@ -500,11 +500,34 @@ impl LogsService for GrpcService {
     }
 }
 
-async fn dispatch_events(events: EventsBuffer, source_context: &SourceContext) {
+async fn dispatch_events(mut events: EventsBuffer, source_context: &SourceContext) {
     if events.is_empty() {
         return;
     }
 
+    if events.has_event_type(EventType::Log) {
+        let mut logs_buffer = FixedSizeEventBuffer::<1024>::default();
+        for log in events.extract(Event::is_log) {
+            // If the buffer is full, dispatch current batch and start a new one
+            if let Some(overflow) = logs_buffer.try_push(log) {
+                if let Err(e) = source_context.dispatcher().dispatch_named("logs", logs_buffer).await {
+                    error!(error = %e, "Failed to dispatch logs")
+                }
+                let mut next = FixedSizeEventBuffer::<1024>::default();
+                next.try_push(overflow);
+                logs_buffer = next;
+                break;
+            }
+        }
+        if !logs_buffer.is_empty() {
+            if let Err(e) = source_context.dispatcher().dispatch_named("logs", logs_buffer).await {
+                error!(error = %e, "Failed to dispatch logs")
+            } else {
+                debug!("Dispatched log events");
+            }
+        }
```

Comment:
We can replace this with a buffered dispatcher:

```rust
let mut buffered_dispatcher = source_context.dispatcher()
    .buffered_named("logs")
    .expect("logs output should exist");

for log_event in events.extract(Event::is_log) {
    if let Err(e) = buffered_dispatcher.push(log_event).await {
        error!(error = %e, "Failed to dispatch log(s).");
    }
}

if let Err(e) = buffered_dispatcher.flush().await {
    error!(error = %e, "Failed to flush log(s).");
}
```

`BufferedDispatcher` handles all of the logic of allocating an event buffer when needed, flushing when full, etc.